types/etc.go

```diff
@@ -189,6 +189,15 @@ func TypeToStr(tp byte, cs string) (r string) {
 	return ts
 }
 
+// TypeToStrForErr converts a field to a string for the error messages.
+func TypeToStrForErr(tp byte) string {
+	ts := type2Str[tp]
+	if tp == mysql.TypeLong {
```

Comment:
what about mysql.TypeLonglong

Reply:
Actually, I don't know the exact map from the type to its name in all MySQL error message. I think we can add more types when we meet the other compatibility issues in future.

Reply:
```
mysql> create table t (i bigint);
Query OK, 0 rows affected (0.02 sec)
mysql> insert into t values ('a');
ERROR 1366 (HY000): Incorrect integer value: 'a' for column 'i' at row 1
```

Reply:
OK, I'll try all the types of MySQL.